packages/synapse-react-client/src/components/DataGrid/MergeGridWithSourceTableButton.tsx

```diff
@@ -0,0 +1,78 @@
+import useMergeGridWithSource from '@/components/DataGrid/useMergeGridWithSource'
+import { displayToast } from '@/components/index'
+import { Button } from '@mui/material'
+import {
+  EntityUpdateResults,
+  instanceOfEntityUpdateResults,
+  instanceOfUploadToTableResult,
+  TableUpdateTransactionResponse,
+} from '@sage-bionetworks/synapse-client'
+
+export type MergeGridWithSourceTableButtonProps = {
+  sourceEntityId: string
+  gridSessionId: string
+}
+
+export default function MergeGridWithSourceTableButton(
+  props: MergeGridWithSourceTableButtonProps,
+) {
+  const { sourceEntityId, gridSessionId } = props
+
+  const { mutate: mergeGridWithSource, isPending } = useMergeGridWithSource({
+    onSuccess: result => {
+      onMergeSuccess(result)
+    },
+    onError: e => {
+      displayToast(e.message, 'danger')
+    },
+  })
+
+  return (
+    <Button
+      loading={isPending}
+      onClick={() => {
+        mergeGridWithSource({ gridSessionId, sourceEntityId })
+      }}
+      variant="contained"
+    >
+      Update table with changes
+    </Button>
+  )
+}
+
+function onMergeSuccess(result: TableUpdateTransactionResponse) {
```

Comment:
After merging, we will also want to give the user the opportunity to delete the grid (or re-sync it when we eventually add that functionality). The grid won't be useful anymore (at least for views) once this happens since the entity will change.